src/Scripting/CSharp/CSharpScript.cs

```diff
@@ -195,15 +200,34 @@ protected override Compilation CreateCompilation()
 
             var references = this.GetReferencesForCompilation();
 
-            var parseOptions = this.Options.IsInteractive ? s_defaultInteractive : s_defaultScript;
-            var tree = SyntaxFactory.ParseSyntaxTree(this.Code, parseOptions, path: this.Path);
+            SyntaxTree[] trees;
+            var diagnostics = DiagnosticBag.GetInstance();
+            try
+            {
+                trees = GetSyntaxTrees(
+                    this.Code,
+                    this.Path,
+                    this.Options.IsInteractive,
+                    diagnostics,
+                    SourceFileResolver.Default,
+                    CSharpScript<T>.DefaultFileReader);
+                if (diagnostics.HasAnyErrors())
+                {
+                    CompilationError(diagnostics);
+                }
+            }
+            finally
+            {
+                diagnostics.Free();
+            }
+            Debug.Assert(trees != null && trees.Length > 0);
```

Comment:
Consider splitting into two asserts, so it's obvious which condition failed when the assert fails.